### PR TITLE
Refactor core for a more streamlined API for aggregators/controllers 

### PR DESCRIPTION
**Pull request by ahellander**
## Description
This is a major refactor of the core library, with the goals to provide: 
 - A clearer plugin-architecture for user- defined helpers and aggregators 
 - a new controller accommodating stragglers and policies for how to handle them  
 - Improvements to the db-backend  schema and tracing
 - Simplification of deployments 
 - Many code and docs improvements

### CODE REVIEW

**Comment by Wrede** [comment]
It's a bit confusing that we have /network and common/net

**Comment by Wrede** [comment]
I think I would move up (towards root) everything in fedn/fedn so that one would have "cli" on the same level as "network" etc. This is to avoid the ugly full paths such as fedn/fedn/fedn/network.

**Comment by ahellander** [comment]
> any reason why reducer don't have it's own folder? Combiner, Clients etc have it...

No particular reason.

File: examples/mnist-keras/README.md [hunk 1]
```diff
@@ -66,3 +66,25 @@ Finally, you can start the experiment from the "control" tab of the UI.
 
 ## Clean up
 You can clean up by running `docker-compose down`.
+<<<<<<< HEAD
```
Comment (Wrede) [inline examples/mnist-keras/README.md:69]: missed resolved conflict here

**Review by Wrede — APPROVED** [review]
the docs need to be modified according to these changes

**Pull request merged**